src/components/Container/Container.tsx

```diff
@@ -0,0 +1,110 @@
+type WrapOptions = "nowrap" | "wrap" | "wrap-reverse";
+
+export interface ContainerProps extends HTMLAttributes<HTMLDivElement> {
+  alignItems?: AlignItemsOptions;
+  children?: React.ReactNode;
+  fillWidth?: boolean;
+  gap?: GapOptions;
+  grow?: GrowShrinkOptions;
+  shrink?: GrowShrinkOptions;
+  hasBorder?: boolean;
+  isResponsive?: boolean;
+  justifyContent?: JustifyContentOptions;
+  orientation?: Orientation;
+  padding?: PaddingOptions;
+  wrap?: WrapOptions;
+}
+
+const Container = ({
+  alignItems = "center",
+  children,
+  fillWidth = false,
+  gap = "none",
+  grow = "1",
+  shrink = "0",
+  hasBorder,
+  isResponsive,
+  justifyContent = "start",
+  orientation = "horizontal",
+  padding = "md",
+  wrap = "nowrap",
+  ...props
+}: ContainerProps) => {
+  return (
+    <Wrapper
+      $alignItems={alignItems}
+      $fillWidth={fillWidth}
+      $gapSize={gap}
+      $grow={grow}
+      $shrink={shrink}
+      $hasBorder={hasBorder}
+      $isResponsive={isResponsive}
+      $justifyContent={justifyContent}
+      $orientation={orientation}
+      $paddingSize={padding}
+      $wrap={wrap}
+      data-testid="container"
+      {...props}
+    >
+      {children}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div<{
+  $alignItems: AlignItemsOptions;
+  $fillWidth?: boolean;
+  $gapSize: GapOptions;
+  $grow: GrowShrinkOptions;
+  $shrink: GrowShrinkOptions;
+  $hasBorder?: boolean;
+  $isResponsive?: boolean;
+  $justifyContent: JustifyContentOptions;
+  $orientation: Orientation;
+  $paddingSize: PaddingOptions;
+  $wrap: WrapOptions;
+}>`
+  display: flex;
+  flex-grow: ${({ $grow = "1" }) => ($grow === "1" ? "1" : `${$grow}`)};
+  flex-shrink: ${({ $shrink = "0" }) => ($shrink === "0" ? "0" : `${$shrink}`)};
+  flex-wrap: ${({ $wrap = "nowrap" }) => ($wrap === "nowrap" ? "nowrap" : `${$wrap}`)};
+  gap: ${({ theme, $gapSize }) => theme.click.container.gap[$gapSize]};
+  padding: ${({ theme, $paddingSize }) => theme.click.container.space[$paddingSize]};
+  background-color: ${({ theme }) => theme.click.global.color.background.muted};
+  width: ${({ $fillWidth = true }) => ($fillWidth === true ? "100%" : "auto")};
+  flex-direction: ${({ $orientation = "horizontal" }) =>
+    $orientation === "horizontal" ? "row" : "column"};
+  align-items: ${({ $alignItems = "center" }) =>
+    $alignItems === "center" ? "center" : `${$alignItems}`};
+  justify-content: ${({ $justifyContent = "left" }) =>
+    $justifyContent === "start" ? "start" : `${$justifyContent}`};
+
+  /* Will remove before merging */
+  border: ${({ $hasBorder }) => ($hasBorder ? "1px solid red" : "none")};
+
+  @media (max-width: 768px) {
+    width: ${({ $isResponsive = true, $fillWidth = true }) =>
+      $isResponsive === true ? "100%" : $fillWidth === true ? "100%" : "auto"};
+    flex-direction: ${({ $isResponsive = true }) =>
+      $isResponsive === true ? "column" : "auto"};
+  }
+`;
+
+export { Container };
```

Comment:
lets also expose 2 simple wrappers, VerticalList and HorizontalList, that populate the orientation by default and improve readability across the code.